docs/docsite/rst/dev_guide/developing_modules_in_bulk.rst

```diff
@@ -0,0 +1,156 @@
+
+On :doc:`community` you can find how to:
+
+* Subscribe to the Mailing Lists - We suggest "Ansible Development List" (for codefreeze info) and "Ansible Announce list"
+* `#ansible-devel` - We have found that IRC `#ansible-devel` works best for module developers so we can have an interactive dialogue.
+* Join the various weekly IRC meetings
+
+
+Your First PR
+``````````````
+
+Assuming you've been through the rest of this document, and not just skipped ahead, you should now be ready to raise your first PR.
+
+The first PR is slightly different to the rest as:
+
+* it defines the namespace
+* it provides a bases for detailed review that will help shape your future PRs
+* it may include shared documentation (`docs_fragments`) that multiple modules require
+* it may include shared code (`module_utils`) that multiple modules require
+
+
+The first PR should include the following files:
+
+* ``lib/ansible/modules/$area/$prefix/__init__.py`` - An empty file to initialize namespace and allow Python to import the files. *Required new file*
+* ``lib/ansible/modules/$area/$prefix/$yourfirstmodule.py`` - A single module. *Required new file*
+* ``lib/ansible/utils/module_docs_fragments/$prefix.py`` - Code documentation, such as details regarding common arguments. *Optional new file*
+* ``lib/ansible/module_utils/$prefix.py`` - Code shared between more than one module, such as common arguments. *Optional new file*
+*  ``docs/docsite/rst/dev_guide/developing_module_utilities.rst`` - Document your new `module_utils` file. *Optional update to existing file*
+
+And that's it.
+
+Before pushing your PR to GitHub it's a good idea to review the :doc:`developing_modules_checklist` again
+
+After publishing your PR on https://github.com/ansible/ansible a Shippable CI test should run (generally within a few minutes), check the results (at the end of the PR page) and ensure it's passing (green), if not inspect each of the results. Most of the errors should be self explanatory and are generally related to badly formatted documentation (see :doc:`YAMLSyntax`) or code that isn't valid Python 2.4 & Python 2.6 (see :doc:`developing_modules_python3`). If you aren't sure what a Shippable test message means copy it into the PR and add as a comment and we will review.
+
+If you need further advice join the `#ansible-devel` IRC channel (details in "Where to get support")
+
+
+We have a "ansibot" helper that comments on GitHub Issues and PRs which should highlight important information.
+
+
+Subsequent PRs
+``````````````
+
+By this point you first PR that defined the module namespace should have been merged. You can take the lessons learned from the first PR and apply it to the rest of the modules.
+
+Raise exactly one PR per module for the remaining modules.
+
+Over the years we've experimented with different sized PRs, some containing one module, some containing five, some even containing many tens of modules in, we've found the following:
```

Comment:
"Over the years we've experimented with different sized module PRs, ranging from one module to many tens of modules, and during that time we've found the following:"